### ISSUE TITLE
Quickstart guide for DOSEMU (not 2) current users?

### ISSUE THREAD

**Issue by Tubatstuff** [stsp/dosemu2#994]
I'm not a game player; but rather a retired DOS/Windows developer.   As a consequence, I have over 20 years of code that I used to host on a Windows XP system.  The original DOSEMU did surprisingly well, and then things started breaking as it was overcome by Linux kernel updates.   That's understandable.   So now I have a drive C: image, complete with FreeDOS, development tools, utilities and batch files from DOSEMU.   My private data resides in my Linux home directory and things are configured so that this is my drive D:.

I'd like to take advantage of DOSEMU2; but it's very confusing.   There already appears to be a drive D: (and F:) and the AUTOEXEC.BAT file on my C: drive is no longer executed on startup.  

I'm sure that all of this can be resolved, but I'm at a loss where to start.   Do you have a "HOWTO" migrate from DOSEMU to DOSEMU2?   In particular, what parts of DOSEMU have to be purged to get things to operate correctly would be useful.

Thanks,

**Comment by jharrison03** [comment]
DOSEmu is setup to work with FreeDOS "out of the box", which is what that drive mapping is about.

You can point DOSemu to your existing DOS by editing the drives in your dosemurc.

**Comment by Tubatstuff** [comment]
Well, see, that's just the thing--no dox.   I purged dosemu2/dosemu and removed all files and directories (including my C: drive) from my system.  Verified with a find / -iname "*dosemu*".

Re-installed dosemu2:
   sudo apt install dosemu2
started dosemu2, got the default empty C: drive as expected as well as D: E: and F: already populated.

No *dosemurc* to be found anywhere on the system.   Where does it (even a skeleton specimen) come from?  No information in /usr/share/doc files on how to boot from a C: drive.

Can you see how hard this would be to someone using dosemu2 for the first time?

**Comment by jschwartzenberg** [comment]
Yes, documentation is still limited. I think the intention was that DOSEMU2 picks up original DOSEMU installations automatically. With all of that in place, doesn't it work?

**Comment by stsp** [comment]
@andrewbird asked me to put the bounty
to that ticket (instead of sending him the bounty
for NDN support), so I do:
Anyone who writes the good quickstart+migration
guide in ML will get 100$. :)
Initial topics to cover, were listed here:
https://github.com/stsp/dosemu2/issues/822#issuecomment-507071398

> I'd like to take advantage of DOSEMU2; but it's very confusing. There already appears
> to be a drive D: (and F:) and the AUTOEXEC.BAT file on my C: drive is no longer
> executed on startup.

You can find most of the explanations in the dosemu.conf
itself, see option `$_hdimage`. You just need to put your
drives between `+0` and `+1` to get your old setup.

**Comment by Tubatstuff** [comment]
Does DOSEMU2 include a sample dosemu.conf file?   It's doesn't seem to be there in the binary distribution (apt) package.

**Comment by stsp** [comment]
Of course there is.
/etc/dosemu/dosemu.conf

**Comment by Tubatstuff** [comment]
Okay, I missed it the first time.

**Comment by stsp** [comment]
I updated the comments in dosemu.conf with
an example for your case (and made it overall
simpler to read).
The doc is still not written exactly because it
is very easy to migrate from dosemu1, and
for us it is obvious (but any small detail may
be non-obvious for a user). We need to collect
the questions like this just to understand what
is to write into the doc.

**Comment by stsp** [comment]
> I think the intention was that DOSEMU2 picks up original DOSEMU installations automatically.

No, that's no longer true.
That was the case initially (in pre-alphas), but
now there are many small details. Only the
default dosemu1 setup can be taken as-is.
If it was modified (eg by playing games with
drive D), then dosemu2 will likely not boot it
in exactly the same way the user expects.

**Comment by stsp** [comment]
I wanted to create the wiki page with the
list of subjects that we may need to document.
But it seems wikis here do not allow any
discussions?
For example there is an old page of @bolle732 
https://github.com/stsp/dosemu2/wiki/Idea-and-wish-list-of-windowed-and-full-screen-modes
to which I don't know how to comment.

So it seems, the suitable work-flow is to create
the ticket per every wiki page, and have the
discussions in such tickets?

For now I've updated the list here: https://github.com/stsp/dosemu2/issues/822#issuecomment-507071398

**Comment by Tubatstuff** [comment]
Okay, Stas,

So far, so good.   Thank you for the updated dosemu.conf.  I'm successfully booting from my C: drive image (a little cleanup necessary).

Can I safely delete the redirection to D:-G: after booting?

C: HD1, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
D: HD2, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
E: HD3, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
F: HD4, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB
G: HD5, Pri[ 1], CHS=    0-1-1, start=     0 MB, size=  2000 MB

--Chuck

**Comment by stsp** [comment]
You dont have to.
The comments clearly say that you
can map those to any drives.
Map something to D-G and move
dosemu2 drives to H etc.

**Comment by stsp** [comment]
I added the ability to create "holes"
in the drive letters. Now you can do:
`$_hdimage = "+0 -2 +1"
to skip D and E at boot, and redirect
them later. See comments in dosemu.conf
for details.

**Comment by emendelson** [comment]
Here's an attempt at a quick-start guide:

https://github.com/dosemu2/dosemu2/issues/1145

**Comment by stsp** [comment]
Well, this ticket is about migration guide
from dosemu1, not a quickstart guide
from zero.

**Comment by emendelson** [comment]
You're right- apologies for wasting bandwidth!
